packages/build-utils/src/clone-env.ts

```diff
@@ -16,14 +16,19 @@ export function cloneEnv(...envs: (Env | undefined)[]): Env {
       return obj;
     }
 
-    // the system path is called `Path` on Windows and Node.js will
-    // automatically return the system path when accessing `PATH`,
-    // however we lose this proxied value when we destructure and
-    // thus we must explicitly copy it
-    if (hasOwnProperty.call(env, 'PATH') || hasOwnProperty.call(env, 'Path')) {
-      obj.PATH = env.PATH;
+    // mixin the env first
+    obj = Object.assign(obj, env);
+
+    if (hasOwnProperty.call(env, 'Path')) {
```

Comment:
**question (non-blocking):** Should we only do this on Windows?

Now that it deletes, it's not really a no-op for other platforms. But maybe we'd rather have it work the same for all platforms anyway--no one gets to access `Path`.

Reply:
It should be harmless for all platforms. While it's true that `Path` should only exist on Windows, there's nothing stopping someone from setting `process.env.Path` as `process.env` is mutable.

Not removing `Path` allowing both `Path` and `PATH` is ultimately what causes the issue. I suspect that `spawn()` will happily pass both `Path` and `PATH` along to the system call, but Win32 will use `Path` or `PATH`, meanwhile if `yarn` looks at the path, it will do the same as `vercel` and look for `PATH`.

Reply:
If we need to remove `Path`, we need to remove it, sure. I'm wondering if there would be other issues caused by it being forcibly deleted when that wasn't happening before.

On Windows, it doesn't seem like it would matter. On another OS, it would only matter if you were explicitly expecting `Path` to show up somewhere, which seems unlikely.